eZ/Publish/Core/FieldType/Tests/RelationListTest.php

```diff
@@ -718,6 +734,77 @@ public function provideInvalidDataForValidate()
         ];
     }
 
+    public function testValidateNotExistingContentRelations(): void
+    {
+        $destinationContentId = (int) 'invalid';
+        $destinationContentId2 = (int) 'invalid-second';
+
+        $this->targetContentValidator
+            ->expects(self::exactly(2))
+            ->method('validate')
+            ->withConsecutive([$destinationContentId], [$destinationContentId2])
+            ->willReturnOnConsecutiveCalls(
+                $this->generateValidationError($destinationContentId),
+                $this->generateValidationError($destinationContentId2)
+            );
+
+        $validationErrors = $this->doValidate([], new Value([$destinationContentId, $destinationContentId2]));
+
+        self::assertIsArray($validationErrors);
+        self::assertCount(2, $validationErrors);
+    }
+
+    public function testValidateInvalidContentType(): void
+    {
+        $destinationContentId = 12;
+        $destinationContentId2 = 13;
+        $allowedContentTypes = ['article', 'folder'];
+
+        $this->targetContentValidator
+            ->expects(self::exactly(2))
```

Comment:
```suggestion
```

We don't care about number of `validate` invocations. Validation result is important here.